scheduler/pkg/agent/repository/mlserver/settings_test.go

```diff
@@ -0,0 +1,265 @@
+			modelSpec: &scheduler.ModelSpec{
+				Requirements: []string{"lightgbm"},
+			},
+			expected: &ModelSettings{
+				Implementation: "mlserver_lightgbm.LightGBMModel",
+				Parameters: &ModelParameters{
+					Uri: "./model.bst",
+				},
+			},
+		},
+		{
+			name:         "xgboost json, top folder, ok",
+			fileToCreate: "model.json",
+			modelSpec: &scheduler.ModelSpec{
+				Requirements: []string{"xgboost"},
+			},
+			expected: &ModelSettings{
+				Implementation: "mlserver_xgboost.XGBoostModel",
+				Parameters: &ModelParameters{
+					Uri: "./model.json",
+				},
+			},
+		},
+		{
+			name:         "joblib, top folder, ok",
+			fileToCreate: "model.joblib",
+			modelSpec: &scheduler.ModelSpec{
+				Requirements: []string{"sklearn"},
+			},
+			expected: &ModelSettings{
+				Implementation: "mlserver_sklearn.SKLearnModel",
+				Parameters: &ModelParameters{
+					Uri: "./model.joblib",
+				},
+			},
+		},
+		{
+			name:         "joblib, sub folder, not ok",
+			fileToCreate: "/tt/model.joblib",
+			modelSpec: &scheduler.ModelSpec{
+				Requirements: []string{"sklearn"},
+			},
+			err: true,
+		},
+		{
+			name:         "pickle, top folder, ok",
+			fileToCreate: "model.pickle",
+			modelSpec: &scheduler.ModelSpec{
+				Requirements: []string{"sklearn"},
+			},
+			expected: &ModelSettings{
+				Implementation: "mlserver_sklearn.SKLearnModel",
+				Parameters: &ModelParameters{
+					Uri: "./model.pickle",
+				},
+			},
+		},
+		{
+			name:         "pkl, top folder, ok",
+			fileToCreate: "model.pkl",
+			modelSpec: &scheduler.ModelSpec{
+				Requirements: []string{"sklearn"},
+			},
+			expected: &ModelSettings{
+				Implementation: "mlserver_sklearn.SKLearnModel",
+				Parameters: &ModelParameters{
+					Uri: "./model.pkl",
+				},
+			},
+		},
+		{
+			name: "unknown, top folder, not ok",
+			modelSpec: &scheduler.ModelSpec{
+				Requirements: []string{"sklearn"},
+			},
+			fileToCreate: "model.foo",
+			err:          true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			path := t.TempDir()
+			folderPath := filepath.Join(path, test.pathToCreate)
+			err := os.MkdirAll(folderPath, fs.ModePerm)
+			g.Expect(err).To(BeNil())
+			if test.fileToCreate != "" {
+				artifactFilePath := filepath.Join(folderPath, test.fileToCreate)
+				err := os.MkdirAll(filepath.Dir(artifactFilePath), fs.ModePerm)
+				g.Expect(err).To(BeNil())
+				err = os.WriteFile(artifactFilePath, []byte{}, fs.ModePerm)
+				g.Expect(err).To(BeNil())
+			}
+			err = createSettingsFile(path, test.modelSpec)
+			if test.err {
+				g.Expect(err).ToNot(BeNil())
+			} else {
+				g.Expect(err).To(BeNil())
+				mlserverHandler := NewMLServerRepositoryHandler(logrus.New())
+				ms, err := mlserverHandler.loadModelSettingsFromFile(filepath.Join(path, mlserverConfigFilename))
+				g.Expect(err).To(BeNil())
+				g.Expect(ms).To(Equal(test.expected))
+			}
+		})
+	}
+}
+
+func TestCreateSKLearnModelSettings(t *testing.T) {
```

Comment:
we are also testing sklearn models above, why this test is different? should we have all sklearn test cases here?

Reply:
I think instead will add more tests for each type of artifact